pkg/types/defaults/installconfig.go

```diff
@@ -26,10 +26,12 @@ func SetInstallConfigDefaults(c *types.InstallConfig) {
 	if c.Networking == nil {
 		c.Networking = &types.Networking{}
 	}
-	if c.Networking.MachineCIDR == nil {
-		c.Networking.MachineCIDR = defaultMachineCIDR
+	if len(c.Networking.MachineNetwork) == 0 {
+		c.Networking.MachineNetwork = []types.MachineNetworkEntry{
+			{CIDR: *defaultMachineCIDR},
+		}
 		if c.Platform.Libvirt != nil {
-			c.Networking.MachineCIDR = libvirtdefaults.DefaultMachineCIDR
+			c.Networking.MachineNetwork[0].CIDR = *libvirtdefaults.DefaultMachineCIDR
```

Comment:
if it's possible that `MachineNetworkEntry` will have more fields in the future then this presumably should be overriding the entire value of `c.Networking.MachineNetwork[0]`, not just `c.Networking.MachineNetwork[0].CIDR`.

Reply:
ack